website_sale_sort_variants_by_availability/models/product_template_attribute_line.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: 2022 Coop IT Easy SCRLfs
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from odoo import models
+
+
+class ProductTemplateAttributeLine(models.Model):
+    _inherit = "product.template.attribute.line"
+
+    def _set_product_template_value_ids(self):
```

Comment:
Because the `sale.variants` XML loops over this:

```xml
                            <t t-foreach="variant_id.product_template_value_ids" t-as="value_id">
```

Which is defined as:

```python
    product_template_value_ids = fields.Many2many(
        'product.template.attribute.value',
        string='Product Attribute Values',
        compute="_set_product_template_value_ids",
        store=False)
```

It's a compute function.

Instead of altering the compute function, we could change the XML to call a custom `get_sorted_product_template_value_ids` or something.